internal/gps/source_cache.go

```diff
@@ -59,6 +59,9 @@ type singleSourceCache interface {
 	// If the input is a revision and multiple UnpairedVersions are associated
 	// with it, whatever happens to be the first is returned.
 	toUnpaired(v Version) (UnpairedVersion, bool)
+
+	// close closes the cache and any backing resources.
+	close() error
```

Comment:
Embed an `io.Closer` in the interface instead?

Reply:
yeah let's please keep this unexported and not worry about`io.Closer()`